### PR TITLE
Build CLI without clap's load_yaml. Fixes #880

### LINKED ISSUE #880: Remove yaml CLI.
There are good reasons to remove it: https://github.com/clap-rs/clap/issues/3087

### PR DESCRIPTION
**Pull request by ddelemeny**
### Description
Replace clap's load_yaml/from(yaml) with command builder functions that call clap's builder api.
NOTE: cli.yaml still exists in the repo for reference, to be deleted once approved and merged

### How was this PR tested?

```
$ cargo test
running 15 tests
test source::tests::test_flatten_json ... ok
test source::tests::test_make_list_sources_table ... ok
test source::tests::test_parse_describe_source_args ... ok
test source::tests::test_parse_delete_source_args ... ok
test source::tests::test_make_describe_source_tables ... ok
test source::tests::test_sniff_params ... ok
test split::tests::test_filter_splits ... ok
test service::tests::test_parse_run_searcher_args ... ok
test service::tests::test_parse_run_indexer_args ... ok
test source::tests::test_parse_add_source_args ... ok
test source::tests::test_parse_list_sources_args ... ok
test tests::test_parse_duration_with_unit ... ok
test split::tests::test_parse_split_describe_args ... ok
test split::tests::test_parse_split_extract_args ... ok
test split::tests::test_parse_list_split_args ... ok

test result: ok. 15 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 10 tests
test tests::test_parse_demux_args ... ok
test tests::test_parse_describe_index_args ... ok
test tests::test_parse_merge_args ... ok
test tests::test_parse_ingest_args ... ok
test tests::test_parse_delete_args ... ok
test tests::test_parse_split_describe_args ... ok
test tests::test_parse_create_args ... ok
test tests::test_parse_split_extract_args ... ok
test tests::test_parse_garbage_collect_args ... ok
test tests::test_parse_search_args ... ok

test result: ok. 10 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.02s


running 15 tests
test test_cmd_all_with_s3_localstack_cli ... ignored
test test_cmd_all_with_s3_localstack_internal_api ... ignored
test test_cmd_dry_run_delete_on_s3_localstack ... ignored
test test_cmd_help ... ok
test test_cmd_create_on_existing_index ... ok
test test_cmd_create ... ok
test test_cmd_ingest_on_non_existing_index ... ok
test test_cmd_ingest_on_non_existing_file ... ok
test test_cmd_delete_index_dry_run ... ok
test test_cmd_delete ... ok
test test_cmd_garbage_collect_no_grace ... ok
test test_cmd_search ... ok
test test_all_local_index ... ok
test test_cmd_ingest_simple ... ok
test test_cmd_garbage_collect_spares_files_within_grace_period ... ok

test result: ok. 12 passed; 0 failed; 3 ignored; 0 measured; 0 filtered out; finished in 6.35s


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s


running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

```

# Caveats
- Clap doesn't provide a clean way to have global AND required arguments, therefore there is a lot of repeated args.
- CLI's `version` and `about` text need to exist out of the build_cli function for their used ref to stay alive. This breaks the intended encapsulation, but I haven't found a clean way to solve this problem.

### CODE REVIEW

File: quickwit-cli/src/cli.rs [hunk 1]
```diff
@@ -18,13 +18,22 @@
...
+use crate::source::{build_source_command, SourceCliCommand};
+use crate::split::{build_split_command, SplitCliCommand};
+
+pub fn build_cli<'a>() -> App<'a> {
+    App::new("Quickwit")
+        .subcommand(build_source_command())
+        .subcommand(build_service_command())
+        .subcommand(build_split_command())
+        .subcommand(build_index_command())
+        .setting(AppSettings::DisableHelpSubcommand)
```
Comment (fmassot) [inline quickwit-cli/src/cli.rs:35]: You also need `ArgRequiredElseHelp` at the top level (just run `cargo r` and you will see that it's not very nice)

File: quickwit-cli/src/main.rs [hunk 2]
```diff
@@ -92,12 +91,9 @@ async fn main() -> anyhow::Result<()> {
         env!("GIT_COMMIT_HASH")
     );
 
-    let yaml = load_yaml!("cli.yaml");
```
Comment (fmassot) [inline quickwit-cli/src/main.rs:95]: don't forget to remove `cli.yaml` file

**Review by fmassot — CHANGES_REQUESTED** [review]
Looks good, add 2 remarks. Can you test the CLI with the examples from the docs? https://quickwit.io/docs/current/reference/cli

File: quickwit-cli/src/generate_markdown.rs [hunk 3]
```diff
@@ -127,7 +126,7 @@ fn markdown_for_subcommand(
             println!(
                 "`--{}` {}{}    ", // space is line break
                 arg.get_name(),
-                arg.get_about().unwrap_or_default(),
+                arg.get_help_heading().unwrap_or_default(),
```
Comment (fmassot) [inline quickwit-cli/src/generate_markdown.rs:129]: why this change?

Comment (fmassot) [reply quickwit-cli/src/generate_markdown.rs:129]: ok the api change is a bit misleading

Comment (ddelemeny) [reply quickwit-cli/src/generate_markdown.rs:129]: Yep, about/get_about vanished. Beta APIS are unstable indeed.

Versioned docs for reference :
old: https://docs.rs/clap/3.0.0-beta.5/clap/struct.Arg.html
new: https://docs.rs/clap/3.0.14/clap/struct.Arg.html
